### PR TITLE
Share on-chain payment requests

### LINKED ISSUE #1291: Share on-chain wallet address to receive payments

### PR DESCRIPTION
**Pull request by Restioson**
Fixes #1291.

### CODE REVIEW

**Comment by holzeis** [comment]
Can we close that after also https://github.com/get10101/10101/pull/1332 has been merged?

**Comment by Restioson** [comment]
CI failure is due to unit tests.

Didn't realise the 'update branch' button would create a merge commit. Will rebase to fix

File: mobile/lib/features/wallet/balance_row.dart [hunk 1]
```diff
@@ -48,28 +47,26 @@ class _BalanceRowState extends State<BalanceRow> with SingleTickerProviderStateM
...
+          SvgPicture.asset("assets/Bitcoin_logo.svg"),
+          walletChangeNotifier.onChain(),
+        ),
+      WalletType.stable => (
+          "Synthetic-USD",
+          theme.lightning,
+          SvgPicture.asset("assets/USD_logo.svg"),
+          positionChangeNotifier.getStableUSDAmountInSats()
+        ),
+    };
```
Comment (bonomat) [inline mobile/lib/features/wallet/balance_row.dart]: cool, I didn't know this syntax is possible in dart.

**Review by bonomat — APPROVED** [review]
Simple. Works. 👍

**Comment by bonomat** [comment]
One thing, shall we plug this into the `receive` floating button as well?

File: mobile/lib/features/wallet/balance_row.dart (hunk 1 above)
Comment (Restioson) [reply mobile/lib/features/wallet/balance_row.dart]: Yea, it's really nice :) - called 'switch expressions'

**Comment by Restioson** [comment]
Actually, I wonder if we shouldn't use BIP47 instead? https://github.com/bitcoin/bips/blob/master/bip-0047.mediawiki

**Comment by bonomat** [comment]
> For example, we could even consider merging the two screens and adding a flip toggle to choose between on-chain and off-chain?

The idea was to introduce a separate screen after clicking `Receive/Send`, i.e. 


https://github.com/get10101/10101/assets/224613/035e8a95-0437-4ba2-9bcf-1804f0309d2a

**Comment by Restioson** [comment]
Ah I see, that could definitely make sense.

Maybe we can leave this for a followup to get this functionality in in the meantime?

**Comment by bonomat** [comment]
> Happy to bikeshed this now, though, if there's no real rush toward the feature.

Will you be working on this right away? If so, let's get this PR in and continue with the new screen in a separate ticket.

**Comment by Restioson** [comment]
> Will you be working on this right away? If so, let's get this PR in and continue with the new screen in a separate ticket.

My plan was to work on the sending of payments next

**Comment by bonomat** [comment]
> > Will you be working on this right away? If so, let's get this PR in and continue with the new screen in a separate ticket.
> 
> My plan was to work on the sending of payments next

Then let's create a ticket for it please

**Comment by Restioson** [comment]
Sure, will add this to merge queue then so long

**Comment by Restioson** [comment]
#1411

**Pull request merged**